app/src/components/filterEntities/entitiesGroup/entitiesGroup.jsx

```diff
@@ -0,0 +1,144 @@
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { reduxForm, getFormValues, getFormSyncErrors } from 'redux-form';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { EntitiesSelector } from 'components/filterEntities/entitiesSelector';
+import styles from './entitiesGroup.scss';
+
+const cx = classNames.bind(styles);
+const debounce = (fn, time) => {
+  let timeout;
+  return (...args) => {
+    const functionCall = () => fn.apply(this, args);
+    clearTimeout(timeout);
+    timeout = setTimeout(functionCall, time);
+  };
+};
+const debouncer = debounce((fn) => {
+  fn();
+}, 1000);
+const ENTITIES_FORM = 'entities-form';
+
+@connect((state) => ({
+  entities: getFormValues(ENTITIES_FORM)(state),
+  formSyncErrors: getFormSyncErrors(ENTITIES_FORM)(state),
+}))
+@reduxForm({
+  form: ENTITIES_FORM,
+  validate: (entities) => {
+    const validationObject = {};
+    Object.keys(entities).forEach((entityId) => {
+      const entity = entities[entityId];
+      entity.validationFunc && (validationObject[entityId] = entity.validationFunc(entity));
+    });
+    return validationObject;
+  },
+})
+export class EntitiesGroup extends Component {
+  static propTypes = {
+    initialize: PropTypes.func.isRequired,
+    change: PropTypes.func.isRequired,
+    onChangeOwn: PropTypes.func.isRequired,
+    formSyncErrors: PropTypes.object.isRequired,
+    entitiesSet: PropTypes.array.isRequired,
+    entities: PropTypes.object,
+  };
+  static defaultProps = {
+    entities: {},
+  };
+
+  state = {
+    activeEntities: [],
+  };
+
+  componentWillMount() {
+    const initObject = {};
+    const activeEntities = [];
+    this.props.entitiesSet.forEach((entity) => {
+      entity.active && activeEntities.push(entity.id);
+      initObject[entity.id] = entity;
+    });
+    this.setState({ activeEntities });
+    this.props.initialize(initObject);
+  }
+
+  formatEntity = (valFromState) => valFromState.value;
+
+  parseEntity = (valFromField, fieldId) =>
+    Object.assign({}, this.props.entities[fieldId], { value: valFromField });
+
+  handleActiveEntities = (entityId) => {
+    const entity = this.props.entities[entityId];
+    this.setState({
+      activeEntities:
+        this.state.activeEntities.indexOf(entityId) !== -1
+          ? this.state.activeEntities.filter((id) => id !== entityId)
+          : this.state.activeEntities.concat([entityId]),
+    });
+    this.props.change(entityId, Object.assign({}, entity, { active: !entity.active }));
+  };
+
+  handleChange = () => {
```

Comment:
Use `componentDidUpdate` instead of `onChange` handler

Reply:
fixed